pkg/graveler/settings/manager_test.go

```diff
@@ -95,7 +193,7 @@ func TestUpdateWithLock(t *testing.T) {
 	wg.Add(IncrementCount)
 	for i := 0; i < IncrementCount; i++ {
 		go func() {
-			testutil.Must(t, m.UpdateWithLock(ctx, "example-repo", "settingKey", emptySettings, update))
+			testutil.Must(t, m.Update(ctx, "example-repo", "settingKey", emptySettings, update))
 			wg.Done()
 		}()
```

Comment:
This is flaky by nature in the KV implementation. There's a chance that a certain goroutine will fail to update the settings within the number of retries allocated to it.